src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -0,0 +1,316 @@
+ */
+
+package org.opensearch.security.tools.democonfig;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.dlic.rest.validation.PasswordValidator;
+import org.opensearch.security.dlic.rest.validation.RequestContentValidator;
+import org.opensearch.security.tools.Hasher;
+
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_MIN_LENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX;
+import static org.opensearch.security.user.UserService.generatePassword;
+
+/**
+ * This class updates the security related configuration, as needed.
+ */
+public class SecuritySettingsConfigurer extends Installer {
+
+    /**
+     * Configures security related changes to the opensearch configuration
+     * 1. Checks if plugins is already configuration. If yes, exit
+     * 2. Sets the custom admin password (Generates one if none is provided)
+     * 3. Write the security config to opensearch.yml
+     */
+    public void configureSecuritySettings() {
+        checkIfSecurityPluginIsAlreadyConfigured();
+        updateAdminPassword();
+        writeSecurityConfigToOpenSearchYML();
+    }
+
+    /**
+     * Replaces the admin password in internal_users.yml with the custom or generated password
+     */
+    static void updateAdminPassword() {
+        String ADMIN_PASSWORD = "";
+        String initialAdminPassword = System.getenv("initialAdminPassword");
+        String ADMIN_PASSWORD_FILE_PATH = OPENSEARCH_CONF_DIR + "initialAdminPassword.txt";
+        String INTERNAL_USERS_FILE_PATH = OPENSEARCH_CONF_DIR + "opensearch-security" + File.separator + "internal_users.yml";
+        boolean shouldValidatePassword = environment.equals(ExecutionEnvironment.demo);
+        try {
+            final PasswordValidator passwordValidator = PasswordValidator.of(
+                Settings.builder()
+                    .put(SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, "(?=.*[A-Z])(?=.*[^a-zA-Z\\\\d])(?=.*[0-9])(?=.*[a-z]).{8,}")
+                    .put(SECURITY_RESTAPI_PASSWORD_MIN_LENGTH, 8)
+                    .build()
+            );
+
+            // Read custom password
+            if (initialAdminPassword != null && !initialAdminPassword.isEmpty()) {
+                ADMIN_PASSWORD = initialAdminPassword;
+            } else {
+                File adminPasswordFile = new File(ADMIN_PASSWORD_FILE_PATH);
+                if (adminPasswordFile.exists() && adminPasswordFile.length() > 0) {
+                    try (BufferedReader br = new BufferedReader(new FileReader(ADMIN_PASSWORD_FILE_PATH, StandardCharsets.UTF_8))) {
```

Comment:
Nit: Do we need to add a catch block for this and log the error that there is something wrong with the file reading?

Reply:
good catch. I originally added it as try-with-resources. but I'll add a catch block.